pages/api/recipes/index.js

```diff
@@ -0,0 +1,19 @@
+import Recipe from "@/db/models/Recipe";
+import connect from "@/db/connect";
+export default async function handler(request, response) {
+  try {
+    await connect();
+    if (request.method === "GET") {
+      const recipes = await Recipe.find();
+      return response.status(200).json(recipes);
+    }
+    if (request.method === "POST") {
+      const recipes = request.body;
+      await Recipe.create(recipes);
+      response.status(201).json({ status: "Recipe created." });
+    }
+  } catch (error) {
+    console.log(error);
+    response.status(400).json({ error: error.message });
+  }
```

Comment:
it would be better like this, to be able to catch errors for the respective asynchronous operations:

```suggestion
 await connect();

  if (request.method === "GET") {
    try {
      const recipes = await Recipe.find();
      return response.status(200).json(recipes);
    } catch (error) {
      console.log(error);
      response.status(400).json({ error: error.message });
    }
  }

  if (request.method === "POST") {
    const recipes = request.body;

    try {
      await Recipe.create(recipes);
      response.status(201).json({ status: "Recipe created." });
    } catch (error) {
      console.log(error);
      response.status(400).json({ error: error.message });
    }
  }
```